src/dags/dag.py

```diff
@@ -89,14 +126,27 @@ def concatenate_functions(
             Enforcing the signature has a small runtime overhead.
         set_annotations (bool): If True, we set the annotations of the concatenated
             function using the annotations of the functions that are used to generate
-            the targets. In case of a type mismatch, an error is raised.
-
+            the targets. Type annotations must be strings, else a
+            NonStringAnnotationError is raised. Either enclose the annotations in quotes
+            or add "from __future__ import annotations" at the top of your file. The
+            return annotation of the concatenated function will be a string, tuple,
+            list, or dict, depending on the return type and the number of targets.
+            Importantly, the return annotation is not a valid type annotation, but
+            designed to communicate as much information on the target types as
+            possible.
 
```

Comment:
```suggestion
        Raises
        ------
        AnnotationMismatchError if `set_annotations` is `True` and there are incompatible annotations in the DAG's components. 
        
```

And/or include with the documentation of the `set_annotations` argument before. Should be symmetric to the description of `NonStringAnnotationError`, also with respect to including it in downstream functions' documentation.